packages/babel-parser/src/plugins/flow.js

```diff
@@ -2721,17 +2723,32 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     skipFlowComment(): number | boolean {
-      const ch2 = this.input.charCodeAt(this.state.pos + 2);
-      const ch3 = this.input.charCodeAt(this.state.pos + 3);
+      const { pos } = this.state;
+      let shiftToFirstNonWhiteSpace = 2;
+      while (
+        [charCodes.space, charCodeTab].includes(
+          this.input.charCodeAt(pos + shiftToFirstNonWhiteSpace),
+        )
+      ) {
+        shiftToFirstNonWhiteSpace++;
+      }
+
+      const ch2 = this.input.charCodeAt(shiftToFirstNonWhiteSpace + pos);
+      const ch3 = this.input.charCodeAt(shiftToFirstNonWhiteSpace + pos + 1);
 
       if (ch2 === charCodes.colon && ch3 === charCodes.colon) {
-        return 4; // check for /*::
+        return shiftToFirstNonWhiteSpace + 2; // check for /*::
       }
-      if (this.input.slice(this.state.pos + 2, 14) === "flow-include") {
```

Comment:
Oh this was wrong, the second argument to slice should be an Index and not the length. So flow-include wasn't working till now. Thanks for fixing this.